File: Swap.cs
using System;

class Swap
{
    static void outinplaceSwap<T>(ref T input1 , ref T input2)
    {
		T temp = input1;
		input1 = input2;
		input2 = temp;
	}

	public static void Main()
	{
        int a = 2;
        int b = 5;
        Console.WriteLine("Initial a = " , a);
        Console.WriteLine("Initial b = ", b);
        outinplaceSwap<int>(ref a,ref b);
        Console.WriteLine("Out-In-place Swap : ");
        Console.WriteLine("a = ", a);
        Console.WriteLine("b = ", b);
    }
}

File: swapFunction/swap.cpp
// swap 
#include<iostream>
using namespace std;
template<class T>
void InplaceSwap(T& input1 , T& input2)
{
    input1 = input1 + input2;
    input2 = input1 - input2;  // intput2 = (input1 + input2) - input2
    input1 = input1 - input2;  // intput2 = (input1 + input2) - input1
}
template<class T>
void OutInplaceSwap(T& input1 , T& input2)
{
    T Auxiliary;
    Auxiliary = input1;
    input1 = input2;
    input2 = Auxiliary;
}
int main()
{
    int a = 2;
    int b = 5;
    cout << "Initial a = " << a << endl;
    cout << "Initial b = " << b << endl;
    cout << "In-place Swap : "  << endl;
    InplaceSwap(a,b);
    cout << "a = " << a << endl;
    cout << "b = " << b << endl;
    cout << "Out-In-place Swap : "  << endl;
    OutInplaceSwap(a,b);
    cout << "a = " << a << endl;
    cout << "b = " << b << endl;
    return 0;
}
